avocado/core/nrunner.py

```diff
@@ -471,13 +471,63 @@ def _create_params(self):
 
         return params
 
+    @staticmethod
+    def _get_avocado_version():
+        """Return the Avocado package version, if installed"""
+        version = "unknown.unknown"
+        if PKG_RESOURCES_AVAILABLE:
+            try:
+                version = pkg_resources.get_distribution(
+                    'avocado-framework').version
+            except pkg_resources.DistributionNotFound:
+                pass
+        return version
+
+    def get_env_variables(self):
+        """Get the default AVOCADO_TEST_* environment variables
+
+        This feature was carried from the legacy runner architecture for
+        compatibility purpose.
+        """
+        outputdir = self.runnable.kwargs.get('AVOCADO_TEST_OUTPUTDIR')
+        with open("/tmp/env.txt", "w") as f:
+            f.write(str(outputdir))
+        # create work dir inside the output dir
+        workdir = os.path.join(outputdir, 'workdir')
+        os.makedirs(workdir, exist_ok=True)
+        # create fake log dir inside output dir
+        logdir = os.path.join(outputdir, 'job-results')
+        os.makedirs(logdir, exist_ok=True)
+        # create fake log file inside the logdir
+        logfile = os.path.join(logdir, 'debug.log')
+        open(logfile, 'w').close()
+        # create the avocado environment variable dictionary
+        avocado_test_env_variables = {
+            'AVOCADO_VERSION': self._get_avocado_version(),
+            'AVOCADO_TEST_BASEDIR': os.path.dirname(
+                self.runnable.uri),
+            'AVOCADO_TEST_WORKDIR': workdir,
+            'AVOCADO_TEST_LOGDIR': logdir,
+            'AVOCADO_TEST_LOGFILE': logfile,
```

Comment:
I'm thinking about the use of this variable too... AS IS, it points to the log file that Avocado manages for the test.  But, with the nrunner model, we *only* capture the output produced by the executable, anything else related to the machinery to execute the test won't make into the log file.

One interesting use case, IMO, would be if this file would be preserved by Avocado, "post mortem".  So that executable tests that want to keep logs out of the standard output and error could write to it.  But, even though this looks interesting, I'm not sure if there's an audience for it.

Reply:
> I'm thinking about the use of this variable too... AS IS, it points to the log file that Avocado manages for the test. But, with the nrunner model, we _only_ capture the output produced by the executable, anything else related to the machinery to execute the test won't make into the log file.

Right, that was what I meant in the scrum meetings when I mentioned that some of the variables might not make sense in the current architecture. As we did not reach a consensus, the best effort was to make a compatibility patch, even without making much sense, and open an issue to rethink the environment variables inside the nrunner.

> 
> One interesting use case, IMO, would be if this file would be preserved by Avocado, "post mortem". So that executable tests that want to keep logs out of the standard output and error could write to it. But, even though this looks interesting, I'm not sure if there's an audience for it.

This is what happens in this patch today, but I also agree with the lack of audience.